Lib/platform.py

```diff
@@ -1230,6 +1230,65 @@ def platform(aliased=0, terse=0):
     _platform_cache[(aliased, terse)] = platform
     return platform
 
+### freedesktop.org os-release standard
+# https://www.freedesktop.org/software/systemd/man/os-release.html
+
+# NAME=value with optional quotes (' or "). The regular expression is less
+# strict than shell lexer, but that's ok.
+_os_release_line = re.compile(
+    "^(?P<name>[a-zA-Z0-9_]+)=(?P<quote>[\"\']?)(?P<value>.*)(?P=quote)$"
+)
+# /etc takes precedence over /usr/lib
+_os_release_candidates = ("/etc/os-release", "/usr/lib/os-relesase")
+_os_release_cache = None
+
+
+def _parse_os_release(lines):
+    # These fields are mandatory fields with well-known defaults
+    # in pratice all Linux distributions override NAME, ID, and PRETTY_NAME.
+    info = {
+        "NAME": "Linux",
+        "ID": "linux",
+        "PRETTY_NAME": "Linux",
+    }
+
+    for line in lines:
+        mo = _os_release_line.match(line)
```

Comment:
The distro module uses `shlex.shlex()` to parse os-release.

```
lexer = shlex.shlex(lines, posix=True)
lexer.whitespace_split = True
```

The freedesktop specification says:

> Shell special characters ("$", quotes, backslash, backtick) must be escaped with backslashes, following shell style.

It seems like your code and the distro module parse `\"` and `\'` differently for example.

Reply:
Please give an example.

Reply:
Something like:

```
NAME='Fedora 19 (Schrödinger\'s Cat)'
```

https://fedoraproject.org/wiki/History_of_Fedora_release_names#Fedora_19_.28Schr.C3.B6dinger.27s_Cat.29

Note: I wrote the example manually, it doesn't come from a concrete os-release file.